code/datums/beam.dm

```diff
@@ -32,7 +32,10 @@
 /datum/beam/New(beam_origin,beam_target,beam_icon='icons/effects/beam.dmi',beam_icon_state="b_beam",time=INFINITY,maxdistance=INFINITY,btype = /obj/effect/ebeam)
 	origin = beam_origin
 	target = beam_target
-	max_distance = maxdistance
+	if(maxdistance < 1)
+		stack_trace("An attempt was made to make a beam with a max distance of 0 or negative.")
```

Comment:
shouldnt this crash?

Reply:
Yeah, I only recently learned the difference thanks to Xander.